src/main.rs

```diff
@@ -66,28 +80,29 @@ impl<W: std::io::Write> RockPaperScissorsGame<W>{
         Ok(true)
       },
       shape_played => {
-        let result = self.evaluate_as_left(shape_played, self.computer_shape);
-        dbg!(
-          // "you played {}, they played {}, result = {}",
+        let result = self.evaluate_game_as_left(shape_played, self.computer_shape);
+        writeln!(
+          self.writer,
+          "You played {}, the computer played {}. {}", 
           shape_played, self.computer_shape, result
-        );
+        )?;
         Ok(false)
       }
     }
   }
 
-  fn transform_shape_to_int(&mut self, shape: ShapesInput) -> i32 {
+  fn transform_shape_to_int(&mut self, shape: ShapesInput) -> Option<i32>  {
     match shape {
-      ShapesInput::Rock => 0,
-      ShapesInput::Paper => 1,
-      ShapesInput::Scissors => 2,
-      _ => 9999
+      ShapesInput::Rock => Some(0),
+      ShapesInput::Paper => Some(1),
+      ShapesInput::Scissors => Some(2),
+      _ => None
     }
   }
 
-  fn evaluate_as_left(&mut self, left: ShapesInput, right: ShapesInput) -> RoundState {
-    let left_int = self.transform_shape_to_int(left);
-    let right_int = self.transform_shape_to_int(right);
+  fn evaluate_game_as_left(&mut self, left: ShapesInput, right: ShapesInput) -> RoundState {
+    let left_int = self.transform_shape_to_int(left).unwrap();
```

Comment:
oh. Lepsie, ale still ti tu padne program, ak ti tam niekto vlozi plne validny input. ShapesInput::Unrecognized je validny input podla typingu a ty na nom panic-nes. Malo by ti to napovedat, ze je nieco zle.